contracts/src/v0.8/ccip/applications/PingPongDemo.sol

```
@@ -64,6 +72,25 @@ contract PingPongDemo is CCIPReceiver, OwnerIsCreator {
     }
   }
 
+  /// @notice A function that is responsible for funding this contract.
+  /// The contract can only be funded if it is set as a nop in the target onRamp.
+  /// In case your contract is not a nop you can prevent this function from being called by setting s_fundingRounds=0.
+  function fundPingPong() public {
+    address onRampAddress = Router(getRouter()).getOnRamp(s_counterpartChainSelector);
+
+    // onRamp does not have anything to pay
+    if (EVM2EVMOnRamp(onRampAddress).getNopFeesJuels() == 0) {
+      return;
+    }
+
+    // not enough link to fund the ping pong
+    if (EVM2EVMOnRamp(onRampAddress).linkAvailableForPayment() < 0) {
```

Maybe better to just revert in the function than silently fail